### PR TITLE
Adding ES Lint to deal with situations where SonarCloud is not available for checking frontend code.

### PR DESCRIPTION
**Pull request by nickskliar-davies**
From experience on a recent project:
- SonarCloud was not available, so we attempted SonarLint. SonarLint crashes VS with vue & typescript.
- ESLint was added to work with the frontend.
- Setup is based on the work done in terms of what to turn off etc.

### CODE REVIEW

**Comment by nickskliar-davies** [comment]
@andriikaplanovskyi I don't know why the tests fail here BTW - I haven't changed anything.

File: src/OneBeyond.Studio.Obelisk.WebUI/package.json [hunk 1]
```diff
@@ -50,8 +52,11 @@
     "@babel/preset-typescript": "7.16.7",
     "@faker-js/faker": "6.3.1",
     "@types/webpack-env": "1.16.0",
+    "@typescript-eslint/eslint-plugin": "^5.48.0",
```
Comment (andriikaplanovskyi) [inline src/OneBeyond.Studio.Obelisk.WebUI/package.json:55]: I remember we've agreed we are not using "^" in packages' versions?

File: OneBeyond.Studio.Obelisk.sln [hunk 2]
```diff
@@ -29,6 +29,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OneBeyond.Studio.Obelisk.Do
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OneBeyond.Studio.Obelisk.Workers", "src\OneBeyond.Studio.Obelisk.Workers\OneBeyond.Studio.Obelisk.Workers.csproj", "{3826B005-74AD-48CA-9CF1-C9BFED6A1196}"
 EndProject
+Project("{54A90642-561A-4BB1-A94E-469ADEE60C69}") = "OneBeyond.Studio.Obelisk.WebUI", "src\OneBeyond.Studio.Obelisk.WebUI\OneBeyond.Studio.Obelisk.WebUI.esproj", "{0DAE7742-89CC-440D-B95D-CE4F6F91D624}"
```
Comment (andriikaplanovskyi) [inline OneBeyond.Studio.Obelisk.sln:32]: I though we do not want to include webUI project into the solution. Our goal was to separate front-end and WebAPI in order for different teams to work on them separately.

**Comment by andriikaplanovskyi** [comment]
> @andriikaplanovskyi I don't know why the tests fail here BTW - I haven't changed anything.

I suspect it is because we do not have any domain tests (yet?) in the Obelisk? @fabiomaistro?

File: src/OneBeyond.Studio.Obelisk.WebUI/package.json (hunk 1 above)
Comment (nickskliar-davies) [reply src/OneBeyond.Studio.Obelisk.WebUI/package.json:55]: Oops - I'll go take them out.

File: OneBeyond.Studio.Obelisk.sln (hunk 2 above)
Comment (dcsl-alexis) [reply OneBeyond.Studio.Obelisk.sln:32]: Agree, with the two newer front ends (coming soon) they will be completely separate, so we shouldn't add this in to the main solution.

Comment (fabiomaistro) [reply OneBeyond.Studio.Obelisk.sln:32]: agree

**Comment by fabiomaistro** [comment]
> > @andriikaplanovskyi I don't know why the tests fail here BTW - I haven't changed anything.
> 
> I suspect it is because we do not have any domain tests (yet?) in the Obelisk? @fabiomaistro?

It's failing because of the change that @nickskliar-davies did to include the`WebUI.esproj` into the solution, so it's trying to run tests within that project. Removing it from the solution should fix this too...

**Comment by andriikaplanovskyi** [comment]
> > > @andriikaplanovskyi I don't know why the tests fail here BTW - I haven't changed anything.
> > 
> > 
> > I suspect it is because we do not have any domain tests (yet?) in the Obelisk? @fabiomaistro?
> 
> It's failing because of the change that @nickskliar-davies did to include the`WebUI.esproj` into the solution, so it's trying to run tests within that project. Removing it from the solution should fix this too...

oh, you're right, sorry, I missed the line in the report

**Comment by nickskliar-davies** [comment]
> > > > @andriikaplanovskyi I don't know why the tests fail here BTW - I haven't changed anything.
> > > 
> > > 
> > > I suspect it is because we do not have any domain tests (yet?) in the Obelisk? @fabiomaistro?
> > 
> > 
> > It's failing because of the change that @nickskliar-davies did to include the`WebUI.esproj` into the solution, so it's trying to run tests within that project. Removing it from the solution should fix this too...
> 
> oh, you're right, sorry, I missed the line in the report

That feels quite fragile then? Surely just adding a project into the solution shouldn't immediately attempt a test cycle? I'll happily remove the `esproj` from the solution, but maybe this is something that needs fixing?

File: OneBeyond.Studio.Obelisk.sln (hunk 2 above)
Comment (nickskliar-davies) [reply OneBeyond.Studio.Obelisk.sln]: Removed.

**Review by fabiomaistro — APPROVED** [review]
LGTM

**Pull request merged**